From 2f3893237ecfe28851d4bd1f7fa1c1755a59cde8 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Thu, 2 Nov 2023 13:54:43 +0000
Subject: [PATCH] Fix AES crypto SIGILL on rpi4-64

Disable assembly routines on arm64/rpi4 since it uses
AES crypto extentions and these extentions do not exist in rpi4

This fixes random SIGILL seen in aes_hw_set_encrypt_key() on rpi4-64

Upstream-Status: Inappropriate [Specific to armv8 minus crypto]
Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 third_party/boringssl/BUILD.gn | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/third_party/boringssl/BUILD.gn b/third_party/boringssl/BUILD.gn
index 52e239f..ef92a2f 100644
--- a/third_party/boringssl/BUILD.gn
+++ b/third_party/boringssl/BUILD.gn
@@ -102,7 +102,7 @@ if (enable_rust) {
   }
 }

-if (is_msan) {
+if (is_msan || current_cpu == "arm64") {
   # MSan instrumentation is incompatible with assembly optimizations.
   # BoringSSL's GAS-compatible assembly knows how to detect MSan, but the NASM
   # assembly does not, so we check for MSan explicitly.
